internal/rpm/info.go

```diff
@@ -0,0 +1,220 @@
+	// handle this.
+	defer func() {
+		if r := recover(); r == nil {
+			return
+		}
+		zlog.Warn(ctx).
+			Str("name", i.Name).
+			Strs("basename", basename).
+			Strs("dirname", dirname).
+			Ints32("dirindex", dirindex).
+			Msg("caught panic in filename construction")
+		i.Filenames = nil
+	}()
+	for j := range basename {
+		// We only want '/'-separated paths, even if running on some other,
+		// weird OS. It seems that RPM assumes '/' throughout.
+		name := path.Join(dirname[dirindex[j]], basename[j])
+		if filePatterns.MatchString(name) {
+			// Record the name as a relative path, as that's what we use
+			// everywhere else.
+			i.Filenames = append(i.Filenames, name[1:])
+		}
+	}
+	return nil
+}
+
+// EVR constructs an Epoch-Version-Release string. That is, a NEVRA without the
+// "N" or "A".
+func (i *Info) EVR() string {
+	var b strings.Builder
+	if i.Epoch != 0 {
+		fmt.Fprintf(&b, "%d:", i.Epoch)
+	}
+	b.WriteString(i.Version)
+	b.WriteByte('-')
+	b.WriteString(i.Release)
+	return b.String()
+}
+
+// Hint constructs a string suitable to be use as the "RepositoryHint".
+func (i *Info) Hint() string {
+	v := url.Values{}
+	if i.Digest != "" {
+		switch i.DigestAlgo {
+		case 8:
+			v.Set("hash", "sha256:"+i.Digest)
+		}
+	}
+	if len(i.Signature) != 0 {
+		prd := packet.NewReader(bytes.NewReader(i.Signature))
+		p, err := prd.Next()
+		for ; err == nil; p, err = prd.Next() {
+			switch p := p.(type) {
+			case *packet.SignatureV3:
+				if p.SigType != 0 {
+					continue
+				}
+				v.Add("key", fmt.Sprintf("%016x", p.IssuerKeyId))
+			case *packet.Signature:
+				if p.SigType != 0 || p.IssuerKeyId == nil {
+					continue
+				}
+				v.Add("key", fmt.Sprintf("%016x", *p.IssuerKeyId))
+			}
+		}
+	}
+	return v.Encode()
+}
+
+// ModuleStream reports the module and stream from the full module version.
+//
+// It returns the empty string if the [Info] does not describe a package from a
+// module.
+func (i *Info) ModuleStream() string {
+	if strings.Count(i.Module, ":") <= 1 {
+		return ""
+	}
+	first := true
+	idx := strings.IndexFunc(i.Module, func(r rune) bool {
+		if r != ':' {
+			return false
+		}
+		if first {
+			first = false
+			return false
+		}
+		return true
+	})
+	return i.Module[:idx]
+}
+
+// FilePatterns is a regular expression for *any* file that may need to be
+// recorded alongside a package.
+//
+// The tested strings are absolute paths.
+var filePatterns *regexp.Regexp
+
+func init() {
+	// TODO(hank) The blanket binary pattern is too broad and can miss things.
+	// Long-term, we should add pattern matching akin to [yara] or file(1) as a
+	// plugin mechanism that all indexers can use. That way, the Go indexer
+	// could register a pattern and use a shared filter over the
+	// [fs.WalkDirFunc] while this package (and dpkg, etc) can tell that another
+	// indexer will find those files relevant.
+	//
+	// [yara]: https://github.com/VirusTotal/yara
+	pat := []string{
+		`^.*/[^/]+\.jar$`, // Jar files
```

Comment:
I know this is copied over, but I'm wondering if we should extend this to the other file extensions we support like WAR, EAR, etc?

Reply:
Yes, but maybe let's address that in https://github.com/quay/claircore/pull/1473